share/completions/conda.fish

```diff
@@ -0,0 +1,302 @@
+# Fish shell completions for conda, the package management system
+# https://conda.io/docs
+# Based on `conda help` as per version 4.4.11
+# Advanced completions:
+# - completion of already defined keys for `conda config`
+# - completion of already created environment where possible
+#
+# Note: conda ships with a completion for fish, automatically generated
+# from `conda -h` but it is far cruder than the completions in this file,
+# and it is by nature far too tricky to tweak to get the desired richness.
+# Hence this effort.
+
+# Filter for `complete -n` selecting a command line
+# with the given conda subcommand
+function __fish_conda_needs_command -a cmd
+    # We need to filter out -x and --xx options so as to complete
+    # the arguments of said options
+    test (string match -r '^\w+' -- (commandline -opc))[-1] = $cmd
```

Comment:
I'd be really careful with `test` - it has some surprisingly sharp edges. This'll fail when the command substitution returns nothing, and it effectively does the same thing `__fish_use_subcommand` does.

We do have `__fish_seen_subcommand_from`, which just tests if a given word has been given. Usually I'll do `not __fish_seen_subcommand_from $allcmds` (where `$allcmds` contains all the subcommands that the command accepts).

That's still not perfect, because it'll see something like `git -C log` as using the "log" subcommand instead of the argument to the "-C" option. To do it properly, you need to actually know which options accept arguments - see how the git or systemctl completions do it - kinda complicated, right?

Reply:
First, the completions work with that function of mine. Then, afaiu, neither `__fish_use_subcommand ` nor `__fish_seen_subcommand_from` fit the bill here because that function `__fish_conda_needs_command` is quite specific: it assesses whether the last non-option argument is the required subcommand, which is exactly what is needed for conda. On the contrary, `__fish_use_subcommand` is too broad, as it settles for any non-option argument. So is `__fish_seen_subcommand_from` as it does not discriminate the position on the command-line.

Reply:
Addressed in 8760b98959dcd0dbb7fbc263c4f67b1317d59913

Reply:
Addressed by 10ff253